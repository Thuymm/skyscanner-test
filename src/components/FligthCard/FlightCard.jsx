import React from "react";

import BpkButton from "@skyscanner/backpack-web/bpk-component-button";
import BpkCard from "@skyscanner/backpack-web/bpk-component-card";
import BpkText from "@skyscanner/backpack-web/bpk-component-text";

import STYLES from "./FlightCard.scss";
import arrow from "./right-arrow.svg";

const getClassName = (className) => STYLES[className] || "UNKNOWN";

const FlightCard = (props) => {
  if (props.data != null) {
    return (
      <BpkCard className={getClassName("card")}>
        <div className={getClassName("card__content")}>
          {props.data.legs.map((leg) => {
            const departure_time = new Date(leg.departure_time);
            const arrival_time = new Date(leg.departure_time);
            return (
              <React.Fragment key={leg.id}>
                <div className={getClassName("card__item-left")}>
                  <img
                    className={getClassName("card__image")}
                    src={`https://logos.skyscnr.com/images/airlines/favicon/${leg.airline_id}.png`}
                  />
                  <BpkText className={getClassName("card__time")}>
                    {departure_time.getHours() +
                      ":" +
                      departure_time.getMinutes()}
                    <br />
                    <BpkText className={getClassName("card__port")}>
                      {leg.departure_airport}
                    </BpkText>
                  </BpkText>
                  <img className={getClassName("card__image")} src={arrow} />
                  <BpkText className={getClassName("card__time")}>
                    {arrival_time.getHours() + ":" + arrival_time.getMinutes()}
                    <br />
                    <BpkText className={getClassName("card__port")}>
                      {leg.arrival_airport}
                    </BpkText>
                  </BpkText>
                </div>
                <div className={getClassName("card__item-right")}>
                  <BpkText className={getClassName("card__duration")}>
                    {Math.round(leg.duration_mins / 60) +
                      "h " +
                      (leg.duration_mins % 60)}
                  </BpkText>
                  <BpkText className={getClassName("card__stop")}>
                    {leg.stops == 0 ? (
                      <span className={getClassName("card__green")}>
                        Direct
                      </span>
                    ) : (
                      <span className={getClassName("card__red")}>
                        {leg.stops} stop
                      </span>
                    )}
                  </BpkText>
                </div>
              </React.Fragment>
            );
          })}

          <div className={getClassName("card__footer")}>
            <BpkText className={getClassName("card__price")} tagName="p">
              {props.data.price}
            </BpkText>
            <BpkText className={getClassName("card__agent")} tagName="p">
              {props.data.agent}
            </BpkText>
          </div>
          <BpkButton large>Select</BpkButton>
        </div>
      </BpkCard>
    );
  } else {
    return <span></span>;
  }
};

export default FlightCard;
