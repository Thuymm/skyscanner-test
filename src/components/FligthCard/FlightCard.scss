@import "~bpk-mixins";

.card {
  margin: 10px 0;

  &__content {
    display: flex;
    flex-wrap: wrap;
    row-gap: 8px;
    column-gap: 20px;
    justify-content: center;
    align-items: center;
    align-content: center;
  }

  &__item-left {
    column-gap: 20px;
    display: flex;
    flex-wrap: nowrap;
    align-items: center;
    justify-content: flex-start;
  }

  &__item-right {
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    justify-content: center;
  }

  &__image {
    width: bpk-spacing-lg();
    height: bpk-spacing-lg();
    vertical-align: top;
  }

  &__duration {
    font-size: $bpk-font-size-sm;
    color: rgb(126, 128, 134);
  }

  &__stop {
    font-size: $bpk-font-size-sm;
    color: rgb(126, 128, 134);
  }

  &__port {
    color: rgb(126, 128, 134);
  }

  &__time {
    font-weight: $bpk-font-weight-bold;
    color: rgb(100, 100, 100);
  }

  &__green {
    color: rgb(146, 200, 205);
  }

  &__red {
    color: rgb(210, 39, 149);
  }

  &__price {
    @include bpk-text--xxl;
    font-weight: $bpk-font-weight-book;
  }

  &__agent {
    @include bpk-text-base;
    color: rgb(210, 210, 210);
  }

  &__footer {
    flex: 20%;
    display: flex;
    flex-wrap: wrap;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
  }

  @media (max-width: 480px) and (orientation: portrait) {
    &__content {
      column-gap: 0;
      justify-content: center;
      align-items: center;
      align-content: center;
    }

    &__item-left {
      flex: 80%;
    }

    &__item-right {
      flex: 20%;
    }

    &__footer {
      flex: 40%;
      align-items: flex-start;
    }
  }
}
