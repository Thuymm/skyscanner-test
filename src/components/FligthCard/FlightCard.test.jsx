import React from "react";
import TestRenderer from "react-test-renderer";

import FlightCard from "./FlightCard";

describe("FlightCard", () => {
  it("should render correctly when there is no item", () => {
    const tree = TestRenderer.create(<FlightCard />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render correctly when there is an item", () => {
    const data = {
      id: "it_1",
      legs: [
        {
          id: "leg_1",
          departure_airport: "BUD",
          arrival_airport: "LTN",
          departure_time: "2020-10-31T15:35",
          arrival_time: "2020-10-31T17:00",
          stops: 0,
          airline_name: "Wizz Air",
          airline_id: "WZ",
          duration_mins: 145,
        },
        {
          id: "leg_4",
          departure_airport: "LTN",
          arrival_airport: "BUD",
          departure_time: "2020-11-11T19:45",
          arrival_time: "2020-11-11T21:10",
          stops: 0,
          airline_name: "Wizz Air",
          airline_id: "WZ",
          duration_mins: 145,
        },
      ],
      price: "Â£35",
      agent: "Wizzair.com",
      agent_rating: 9.1,
    };
    const tree = TestRenderer.create(<FlightCard data={data} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
