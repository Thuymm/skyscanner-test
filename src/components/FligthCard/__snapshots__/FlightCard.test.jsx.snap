// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightCard should render correctly when there is an item 1`] = `
<div
  className="bpk-card bpk-card--padded card"
  role="button"
  tabIndex={0}
>
  <div
    className="card__content"
  >
    <div
      className="card__item-left"
    >
      <img
        className="card__image"
        src="https://logos.skyscnr.com/images/airlines/favicon/WZ.png"
      />
      <span
        className="bpk-text bpk-text--body-default card__time"
      >
        15:35
        <br />
        <span
          className="bpk-text bpk-text--body-default card__port"
        >
          BUD
        </span>
      </span>
      <img
        className="card__image"
        src="right-arrow.svg"
      />
      <span
        className="bpk-text bpk-text--body-default card__time"
      >
        15:35
        <br />
        <span
          className="bpk-text bpk-text--body-default card__port"
        >
          LTN
        </span>
      </span>
    </div>
    <div
      className="card__item-right"
    >
      <span
        className="bpk-text bpk-text--body-default card__duration"
      >
        2h 25
      </span>
      <span
        className="bpk-text bpk-text--body-default card__stop"
      >
        <span
          className="card__green"
        >
          Direct
        </span>
      </span>
    </div>
    <div
      className="card__item-left"
    >
      <img
        className="card__image"
        src="https://logos.skyscnr.com/images/airlines/favicon/WZ.png"
      />
      <span
        className="bpk-text bpk-text--body-default card__time"
      >
        19:45
        <br />
        <span
          className="bpk-text bpk-text--body-default card__port"
        >
          LTN
        </span>
      </span>
      <img
        className="card__image"
        src="right-arrow.svg"
      />
      <span
        className="bpk-text bpk-text--body-default card__time"
      >
        19:45
        <br />
        <span
          className="bpk-text bpk-text--body-default card__port"
        >
          BUD
        </span>
      </span>
    </div>
    <div
      className="card__item-right"
    >
      <span
        className="bpk-text bpk-text--body-default card__duration"
      >
        2h 25
      </span>
      <span
        className="bpk-text bpk-text--body-default card__stop"
      >
        <span
          className="card__green"
        >
          Direct
        </span>
      </span>
    </div>
    <div
      className="card__footer"
    >
      <p
        className="bpk-text bpk-text--body-default card__price"
      >
        Â£35
      </p>
      <p
        className="bpk-text bpk-text--body-default card__agent"
      >
        Wizzair.com
      </p>
    </div>
    <button
      className="bpk-button bpk-button--large"
      disabled={false}
      onClick={null}
      type="button"
    >
      Select
    </button>
  </div>
</div>
`;

exports[`FlightCard should render correctly when there is no item 1`] = `<span />`;
